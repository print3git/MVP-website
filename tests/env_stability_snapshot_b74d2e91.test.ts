const critical = [
  "SPARC3D_TOKEN",
  "CLOUDFRONT_MODEL_DOMAIN",
  "STABILITY_KEY",
  "HF_TOKEN",
  "AWS_ACCESS_KEY_ID",
  "AWS_SECRET_ACCESS_KEY",
  "DB_URL",
  "STRIPE_SECRET_KEY",
  "STRIPE_WEBHOOK_SECRET",
];

function snapshotEnv() {
  const out = {};
  for (const key of critical) {
    out[key] = process.env[key] ? "<set>" : undefined;
  }
  return out;
}

describe("env stability snapshot", () => {
  test("matches baseline and required vars present", () => {
    const snap = snapshotEnv();
    console.log("env snapshot", JSON.stringify(snap, null, 2));
    expect(snap.SPARC3D_TOKEN).toBe("<set>");
    expect(snap.CLOUDFRONT_MODEL_DOMAIN).toBe("<set>");
    expect(snap).toMatchSnapshot();
  });
});
