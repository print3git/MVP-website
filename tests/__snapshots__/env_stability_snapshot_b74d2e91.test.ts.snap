// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`env stability snapshot matches baseline and required vars present 1`] = `
{
  "AWS_ACCESS_KEY_ID": "<set>",
  "AWS_SECRET_ACCESS_KEY": "<set>",
  "CLOUDFRONT_MODEL_DOMAIN": "<set>",
  "DB_URL": "<set>",
  "HF_TOKEN": undefined,
  "SPARC3D_TOKEN": "<set>",
  "STABILITY_KEY": undefined,
  "STRIPE_SECRET_KEY": "<set>",
  "STRIPE_WEBHOOK_SECRET": "<set>",
}
`;
